js/stdlib/ts/ALGO.ts

```diff
@@ -1652,15 +1656,55 @@ export const connectAccount = async (networkAccount: NetworkAccount): Promise<Ac
       return { getView1, viewLib };
     };
 
-    const setupEvents = (a: any, b: any) => {
-      void a;
-      void b;
-      const getEvent = async (evt: string, tys: any) => {
-        void evt;
-        void tys;
-        return { when: 0, what: undefined };
-      }
-      return { getEvent };
+    const setupEvents = (a: SetupEventArgs) => {
+      const eventCache = new EventCache();
+      const getC = makeGetC(a, eventCache);
+      const createEventStream = (event: string, tys: AnyALGO_Ty[]) => {
+        let time: BigNumber | null = null;
```

Comment:
I think this time might be part of the same state as the eventCache (i.e. out one level of scope)